src/lib/decodeCalldata.ts

```diff
@@ -15,7 +15,7 @@ export function decodeCalldata(
       decoded = iface.decodeFunctionData(frag.name, calldata);
       const encoded = iface.encodeFunctionData(frag.name, decoded);
       assert(
-        encoded.length === calldata.length,
```

Comment:
i assume that this was just an optimization that backfired... am I missing something? @LogvinovLeon

Reply:
It wasn't intended to be an optimization, but it's better the way it is now.
My thought process here was - decoding/encoding is 1-1, so it's enough to compare length.
Do you have a counterexample for this?

Reply:
Yes. It's in the test case :D